### PR TITLE
Feature/one login integration

### PR DESCRIPTION
**Pull request by dylanwrightCO**

### CODE REVIEW

File: src/main/java/gov/cabinetofice/gapuserservice/service/OneLoginService.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    private String oneLoginBaseUrl;
+
+    @Value("${onelogin.client-assertion-type}")
+    private String clientAssertionType;
+
+    @Value("${onelogin.service-redirect-url}")
+    private String serviceRedirectUrl;
+
+    @Value("${onelogin.private-key}")
+    public String privateKey;
```
Comment (dominicwest) [inline src/main/java/gov/cabinetofice/gapuserservice/service/OneLoginService.java:42]: Could pull these in from a config bean instead, but pretty indifferent on this

File: src/main/java/gov/cabinetofice/gapuserservice/exceptions/PrivateKeyParsingException.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package gov.cabinetofice.gapuserservice.exceptions;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.FORBIDDEN)
```
Comment (ryan-tco) [inline src/main/java/gov/cabinetofice/gapuserservice/exceptions/PrivateKeyParsingException.java:6]: If it's a private key parsing error on the server, is 403 the best response?

**Review by dominicwest — APPROVED** [review]

File: src/main/java/gov/cabinetofice/gapuserservice/exceptions/PrivateKeyParsingException.java (hunk 2 above)
Comment (dylanwrightCO) [reply src/main/java/gov/cabinetofice/gapuserservice/exceptions/PrivateKeyParsingException.java]: Yeah fair point, changed to internal server error.

File: src/main/java/gov/cabinetofice/gapuserservice/service/OneLoginService.java (hunk 1 above)
Comment (dylanwrightCO) [reply src/main/java/gov/cabinetofice/gapuserservice/service/OneLoginService.java:42]: I feel like this way is more explicit and readable

**Review by dominicwest — APPROVED** [review]

**Review by ryan-tco — APPROVED** [review]

**Pull request merged**